library/systemd/src/lib/yast2/system_service.rb

```diff
@@ -0,0 +1,490 @@
+    #
+    # See {#start_modes} to find out the supported modes for a given service (usually :on_boot,
+    # :manual and, in some cases, :on_demand). The given value will be applied after calling {#save}.
+    #
+    # @see #start_modes
+    #
+    # @raise ArgumentError when mode is not valid
+    def start_mode=(mode)
+      if !start_modes.include?(mode)
+        raise ArgumentError, "Invalid start mode: '#{mode}' for service '#{service.name}'"
+      end
+
+      if mode == current_start_mode
+        unregister_change(:start_mode)
+      else
+        register_change(:start_mode, mode)
+      end
+    end
+
+    # Whether the service supports :on_demand start mode
+    #
+    # @return [Boolean]
+    def support_start_on_demand?
+      start_modes.include?(:on_demand)
+    end
+
+    # Whether the service will be active after calling {#save}
+    #
+    # @note This is a temporary value (not saved yet). Use {#current_active?} to get the actual
+    #   active value of the service in the system.
+    #
+    # @return [Boolean] true if the service must be active; false otherwise
+    def active?
+      return new_value_for(:active) if changed?(:active)
+      current_active?
+    end
+
+    # Keywords to search for this service
+    #
+    # In case the service has an associated socket, the socket name
+    # is included as keyword.
+    #
+    # @return [Array<String>] e.g., #=> ["tftp.service", "tftp.socket"]
+    def keywords
+      keywords = [service.id]
+      keywords << socket.id if socket
+      keywords
+    end
+
+    # Sets the service to be started after calling {#save}
+    #
+    # @see #active=
+    #
+    # @return [Symbol] :start
+    def start
+      self.active = true
+      self.action = :start
+    end
+
+    # Sets the service to be stopped after calling {#save}
+    #
+    # @see #active=
+    #
+    # @return [Symbol] :stop
+    def stop
+      self.active = false
+      self.action = :stop
+    end
+
+    # Sets the service to be restarted after calling {#save}
+    #
+    # @return [Symbol] :restart
+    def restart
+      register_change(:active, true)
+      self.action = :restart
+    end
+
+    # Sets the service to be reloaded after calling {#save}
+    #
+    # @return [Symbol] :reload
+    def reload
+      register_change(:active, true)
+      self.action = :reload
+    end
+
+    # Saves changes into the underlying system
+    #
+    # @note Cached changes are reset and the underlying service is refreshed.
+    #
+    # @param keep_state [Boolean] Do not change service status. Useful when running on 1st stage.
+    #
+    # @return [Boolean] true if the service was saved correctly; false otherwise.
+    def save(keep_state: false)
+      clear_errors
+      save_start_mode
+      perform_action unless keep_state
+      reset && refresh
+    end
+
+    # Reverts cached changes
+    #
+    # @return [Boolean] true if the service was reset correctly. Actually, the
+    #   service always can be reset.
```

Comment:
Does this mean this method is a part of a team of methods with the same interface? Use a `@see` tag?